dojo/engagement/signals.py

```diff
@@ -0,0 +1,57 @@
+from auditlog.models import LogEntry
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from django.dispatch import receiver
+from django.db.models.signals import pre_save, post_save, post_delete
+from django.utils.translation import gettext as _
+from django.urls import reverse
+
+from dojo.models import Engagement
+from dojo.notifications.helper import create_notification
+
+
+@receiver(post_save, sender=Engagement)
+def engagement_post_save(sender, instance, created, **kwargs):
+    if created:
+        title = 'Engagement created for ' + str(instance.product) + ': ' + str(instance.name)
+        create_notification(event='engagement_added', title=title, engagement=instance, product=instance.product,
+                            url=reverse('view_engagement', args=(instance.id,)))
+
+
+@receiver(pre_save, sender=Engagement)
+def engagement_pre_save(sender, instance, **kwargs):
+    old = sender.objects.filter(pk=instance.pk).first()
+    if old and instance.status != old.status:
+        if instance.status in ["Cancelled", "Completed"]:
+            create_notification(event='engagement_closed',
+                                title=_('Closure of %s') % instance.name,
+                                description=_('The engagement "%s" was closed') % (instance.name),
+                                engagement=instance, url=reverse('engagement_all_findings', args=(instance.id, )))
+        elif instance.status in ["In Progress"]:
```

Comment:
If an Engagement is "Not Started" and I add a Test, this results in a notification that the Engagement was "reopened". I guess the check here should discard these notifications if `old`'s status was "Not Started"

No Tests under the `asdfadsf` Engagement, status of "Not Started":
![Screenshot 2024-04-09 at 19 56 38](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/321260e8-4150-490e-a5c7-9819a83497f9)

Test added, notification that `asdfadsf` was "reopened":
![Screenshot 2024-04-09 at 19 57 05](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/b6527193-5067-4d85-a56b-4681b324aee4)

View of Engagement after Test was added:
![Screenshot 2024-04-09 at 19 57 12](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/2d35f2ba-c9ec-458f-a9c6-0bc7ac8d84cd)

Reply:
Make sense.
Implemented and covered in unittests as well: https://github.com/DefectDojo/django-DefectDojo/pull/8824/commits/575bb4e99d5714f759c02cbe6d9918388e312b7f